lib/btrfs_test.pm

```diff
@@ -18,13 +19,53 @@ sub set_unpartitioned_disk_in_bash {
         $vd = 'sd';
     }
     assert_script_run 'parted --script --machine -l';
-    assert_script_run 'disk=${disk:-$(parted --script --machine -l |& sed -n \'s@^\(/dev/' . $vd . '[ab]\):.*unknown.*$@\1@p\')}';
+    assert_script_run 'disk=${disk:-$(parted --script --machine -l |& sed -n \'s@^\(/dev/' . $vd . 'b\):.*$@\1@p\')}';
```

Comment:
I am sorry, I don't know what weirdness you mean. `set_unpartitioned_disk` should do find the unpartitioned disk. What you are assuming is that /dev/vdb is always the "playground" disk regardless of content but we can not rely on that.